pulsar-proxy/src/main/java/org/apache/pulsar/proxy/server/ProxyConnection.java

```diff
@@ -481,6 +504,30 @@ public HAProxyMessage getHAProxyMessage() {
         return haProxyMessage;
     }
 
-    private static final Logger LOG = LoggerFactory.getLogger(ProxyConnection.class);
+    private boolean isBrokerActive(String targetBrokerHostPort) {
+        for (ServiceLookupData serviceLookupData : getAvailableBrokers()) {
+            if (matchesHostAndPort("pulsar://", serviceLookupData.getPulsarServiceUrl(), targetBrokerHostPort)
+                    || matchesHostAndPort("pulsar+ssl://", serviceLookupData.getPulsarServiceUrlTls(),
+                    targetBrokerHostPort)) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    private List<? extends ServiceLookupData> getAvailableBrokers() {
+        try {
+            return service.getDiscoveryProvider().getAvailableBrokers();
```

Comment:
This would not work if proxy doesn't have a ZK connection (which is the most common way to configure it). 

(It would actually trigger an NPE here).

Reply:
Thanks @merlimat . I added logging for that and created https://github.com/apache/pulsar-helm-chart/pull/221 to configure the ZK connection in Apache Pulsar Helm chart by default.

Reply:
It's actually better to not have the proxy talk with ZK, mostly for stability issues. That is how it's typically used in production and why it's not set in the helm chart.

Reply:
There has been many cases in which proxies were left in weird state because of ZK sessions or cases in which proxies were having impact on ZK performance. 

For all these reasons, the default has always been to not have the proxy talk directly to ZK, instead relying on DNS service discovery.

Reply:
There's now a revisited solution. @merlimat PTAL